File: LeetCode_1768/main.go
/**
  LeetCode_1768
  https://leetcode.com/problems/merge-strings-alternately/
  MatrixDeity, 2023.
*/

package leetcode1768

import (
	"strings"
)

func mergeAlternately(word1 string, word2 string) string {
	builder := strings.Builder{}
	builder.Grow(len(word1) + len(word2))

	i := 0
	for ; i < len(word1) && i < len(word2); i++ {
		builder.WriteByte(word1[i])
		builder.WriteByte(word2[i])
	}

	for ; i < len(word1); i++ {
		builder.WriteByte(word1[i])
	}

	for ; i < len(word2); i++ {
		builder.WriteByte(word2[i])
	}

	return builder.String()
}

File: LeetCode_1768/main.cpp
/**
    LeetCode_1768
    https://leetcode.com/problems/merge-strings-alternately/
    MatrixDeity, 2023.
*/

#include <string>

using namespace std;

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res;
        auto n = word1.size(), m = word2.size();
        res.reserve(n + m);

        size_t i = 0;
        for (; i < n && i < m; ++i) {
            res.push_back(word1[i]);
            res.push_back(word2[i]);
        }

        for (; i < n; ++i) {
            res.push_back(word1[i]);
        }

        for (; i < m; ++i) {
            res.push_back(word2[i]);
        }

        return res;
    }
};
